### PR TITLE
Yearn plugin scripts

### PR DESCRIPTION
**Pull request by tbrent**
Needs 2 reviewers to go in

### CODE REVIEW

File: scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts
Comment (pmckelvy1) [inline scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:64]: we should run the backtester on this to confirm this much hiding is needed. i have a feeling it could be a couple orders of magnitude lower

**Comment by pmckelvy1** [comment]
lgtm

File: scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts
Comment (tbrent) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:64]: This isn't how I would think about it -- the question is not "what is the largest drawdown that has occurred historically" it is "what is the largest drawdown that might reasonably happen in the future that should not result in the collateral being defaulted".

Comment (akshatmittal) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:64]: AFAIK, Yearn stable vaults are up-only especially the curve vaults since the initial LP is never lost/touched. However, I don't quite remember on how I know this, so maybe it's worth running the backtester anyway to see if there's a drawdown.

Comment (tbrent) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:64]: I searched through the discord when doing my research for this and they indicated it is exceedingly unlikely the rate would fall but it is not up-only. An individual strategy might be up-only, but not all strategies are this way.

**Comment by tbrent** [comment]
TODO: Decide on a revenue hiding value

**Comment by tbrent** [comment]
Decided on:
- Using the new helper
- Using a very small revenue hiding since we can assume underlying curve pool is

**Comment by tbrent** [comment]
@akshatmittal @pmckelvy1 this is ready for y'all; need 2 approvals

File: scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts
Comment (akshatmittal) [inline scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:67]: Did we run a backtest on this?

Comment (tbrent) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:67]: No, I talked to @jankjr and he mentioned the backtest was manual. Let me pull him in and see if we can get it backtested before the merge to `master`

Comment (tbrent) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:67]: Confirmed @jankjr will backtest this and that will block merging this PR

Comment (tbrent) [reply scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts:67]: Backtest successful

**Comment by tbrent** [comment]
@akshatmittal ready for approval

**Review by akshatmittal — APPROVED** [review]
Assuming the addresses in the constants file are correct (unchanged in this PR), the PR looks good!

**Comment by pmckelvy1** [comment]
lgtm, but do we want to put it into master or 3.1.0? either way, we should get it merged up into 3.2.0

**Pull request merged**
